File: Rentify Server/models/Service.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const Service = new Schema({
    landlord_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        require: false
    },
    name: { type: String, require: true },
    description: { type: String, require: true },
    price: { type: Number, require: true },
    photos: { type: [String], required: true },
    created_at: { type: Date, default: Date.now },
    updated_at: { type: Date, default: Date.now }
});

module.exports = mongoose.model("Service", Service);

File: Rentify/app/src/main/java/com/rentify/user/app/model/Model/Service.kt
package com.rentify.user.app.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class Service(
    val name: String,
    val description: String,
    val price: Double,
    val photos: List<String>? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null
): Parcelable

data class ListServiceResponse(
    @SerializedName("_id") val _id: String,
    @SerializedName("created_at")val created_at: String,
    @SerializedName("description")val description: String,
    @SerializedName("landlord_id")val landlord_id: String,
    @SerializedName("name")val name: String,
    @SerializedName("photos")val photos: List<String>,
    @SerializedName("price")val price: Int,
    @SerializedName("updated_at")val updated_at: String
)

